text/0000-minimal-builders.md

```diff
@@ -0,0 +1,246 @@
+
+1. Publish a Github Action, Concourse task, and other pre-made CI/CD tooling
+   templates that use `jam update builder` to keep users' custom builders up to
+   date.
+   - Benefits:
+     - Users can benefit from custom builders with diminished time investment
+   - Drawbacks:
+     - Paketo isn't a CI/CD tooling-focused project. This would expand the scope
+     of released artifacts for which the project is responsible.
+
+1. Make upstream contributions to pack that speed up builder pull times
+   - Benefits:
+     - Addresses the platform's problem of slow build times more directly. Will
+      benefit performance for users with _all_ builders.
+   - Drawbacks:
+     - Requires upstream contribution that may require an RFC. Optimizing image
+    pulls from a registry does not necessarily fall in scope of the Paketo
+    project currently. May require Paketo contributors to build up expertise
+    that is tangential to Paketo's main aims.
+
+<!--- Some other alternatives, deprioritized for now:
+- publish minimal builders, but check in their `builder.toml` files to the same
+  repos where the standard builders are. This increases discoverability of
+  buildpack compatibility. May also increase complexity of automations +
+  maintenance - we'd need to publish 2 github releases to the same repo. Github's support for this is poor.
+- instead of minimal builders, publish builders with curated sets of commonly-combined buildpacks (e.g. builders for each language family buildpack in isolation)
+--->
+
+## Implementation
+
+### Repositories
+Three new repositories should be created, for each of the new minimal builders:
+- paketo-buildpacks/minimal-full-builder
+- paketo-buildpacks/minimal-base-builder
+- paketo-buildpacks/minimal-tiny-builder
+
+
+### Automation
+
+In the above 3 repositories, there should be automation that mimics the
+automation in the existing builder repositories to keep the build image and
+lifecycle versions up-to-date.
+
+### Publishing
+
+The builders should be pushed to Dockerhub and GCR with tags as follows (using Minimal Full Builder as an example):
+- `paketobuildpacks/builder:minimal-full` for the latest builder
+- `paketobuildpacks/builder:minimal-full-1.2.3` for version 1.2.3 of the builder
+
+### Documentation
+The READMEs in the builders' repos, the `description` fields in their
+`builder.toml` files and the Paketo website should all clearly document that
+these builders cannot be used without buildpacks specified at run-time.
+Documentation should also clearly state how a user can establish which
+buildpacks are compatible with which builders (given that the
+`builder.toml`will not otherwise indicate this).
+
+<!---
+## Prior Art
+
+{{This section is optional if there are no actual prior examples in other tools.}}
+{{Discuss existing examples of this change in other tools, and how they've addressed various concerns discussed above, and what the effect of those decisions has been.}}
+--->
+
```

Comment:
Would we want `pack builder suggest` to include these builders. Maybe not because they would provide a surprising/broken first build experience, but also it could help more users become aware of them.

Reply:
My vote would be to not include them, or break them into an "Advanced" grouping or something so they are set apart in the output. If it's not clearly marked, I agree it could be confusing for new users.

Reply:
If we wanted these builders surfaced in `pack builder suggest`, what would be the process for achieving that? An RFC elsewhere?

Reply:
If we wanted it, I think we could make a PR to to pack directly.